extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/ClassConfigPropertiesUtil.java

```diff
@@ -0,0 +1,359 @@
+package io.quarkus.arc.deployment.configproperties;
+
+import static io.quarkus.arc.deployment.configproperties.ConfigPropertiesUtil.ReadOptionalResponse;
+import static io.quarkus.arc.deployment.configproperties.ConfigPropertiesUtil.createReadMandatoryValueAndConvertIfNeeded;
+import static io.quarkus.arc.deployment.configproperties.ConfigPropertiesUtil.createReadOptionalValueAndConvertIfNeeded;
+import static io.quarkus.arc.deployment.configproperties.ConfigPropertiesUtil.determineSingleGenericType;
+
+import java.lang.reflect.Modifier;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.Set;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.config.Config;
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.FieldInfo;
+import org.jboss.jandex.IndexView;
+import org.jboss.jandex.MethodInfo;
+import org.jboss.jandex.Type;
+
+import io.quarkus.deployment.bean.JavaBeanUtil;
+import io.quarkus.deployment.util.HashUtil;
+import io.quarkus.gizmo.BranchResult;
+import io.quarkus.gizmo.BytecodeCreator;
+import io.quarkus.gizmo.ClassCreator;
+import io.quarkus.gizmo.ClassOutput;
+import io.quarkus.gizmo.FieldCreator;
+import io.quarkus.gizmo.FieldDescriptor;
+import io.quarkus.gizmo.MethodCreator;
+import io.quarkus.gizmo.MethodDescriptor;
+import io.quarkus.gizmo.ResultHandle;
+import io.quarkus.runtime.StartupEvent;
+
+final class ClassConfigPropertiesUtil {
+
+    private static final String JAVAX_VALIDATION_PREFIX = "javax.validation";
+    private static final String VALIDATOR_CLASS = "javax.validation.Validator";
+    private static final String PATH = "javax.validation.Path";
+    private static final String CONSTRAINT_VIOLATION_CLASS = "javax.validation.ConstraintViolation";
+    private static final String HIBERNATE_VALIDATOR_IMPL_CLASS = "org.hibernate.validator.internal.engine.ValidatorImpl";
```

Comment:
Please don't use an internal class, better use `org.hibernate.validator.HibernateValidator` which is part of the public API.